File: 2023-2024/sourcecode/mod1/lab3/12JetpackNavigationBasicsJava/app/src/main/java/com/example/a12_jetpacknavigationbasics_java/FragmentA.java
package com.example.a12_jetpacknavigationbasics_java;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavDirections;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.a12_jetpacknavigationbasics_java.databinding.FragmentABinding;

public class FragmentA extends Fragment {

    private FragmentABinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentABinding.inflate(inflater);

        binding.fabA.setOnClickListener(view -> {
            NavDirections action = FragmentADirections.actionFragmentAToFragmentB(5);
            Navigation.findNavController(requireView()).navigate(action);

//            Bundle args = new Bundle();
//            args.putInt("key", 5);
//            Navigation.findNavController(requireView()).navigate(R.id.fragmentB, args);
        });

        return binding.getRoot();
    }
}

File: 2023-2024/sourcecode/mod1a/lab3/11JetpackNavigationBasicsKotlin/app/src/main/java/com/example/a11_jetpacknavigationbasics_kotlin/FragmentA.kt
package com.example.a11_jetpacknavigationbasics_kotlin

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.Navigation
import com.example.a11_jetpacknavigationbasics_kotlin.databinding.FragmentABinding

class FragmentA : Fragment() {

    private lateinit var binding: FragmentABinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentABinding.inflate(inflater)

        binding.fabA.setOnClickListener {
            val action = FragmentADirections.actionFragmentAToFragmentB(5)
            Navigation.findNavController(requireView()).navigate(action)

//            val args = Bundle()
//            args.putInt("key", 5)
//            Navigation.findNavController(requireView()).navigate(R.id.fragmentB, args)
        }

        return binding.root
    }
}